ui/src/components/UserMenu/index.jsx

```diff
@@ -57,9 +57,6 @@ export default class UserMenu extends Component {
   handleSignOutClick = () => {
     this.handleMenuClose();
     this.props.onUnauthorize();
-    // Since Apollo caches query results, it’s important to get rid of them
-    // when the login state changes.
-    this.props.client.clearStore();
```

Comment:
Would calling `this.props.client.resetStore()` here still be a good idea?

Reply:
Re-fetching the active queries won't work for many of our views where we purposely avoid pagination by fetching all the data via (`fetchMore`) on page load. `this.props.client.resetStore()` would only re-fetch the first page for paginated endpoints.

Reply:
So if I understand, the combination of clearing the cache here, re-rendering the `Main` component, and using `network-only` on any secret views means that views will not just have their queries re-run, but will re-render from an initial empty state (so resetting to the first page, for example).  Is that about right?  If we're clearing the cache, why is the `network-only` policy required?

Reply:
> If we're clearing the cache

We're not clearing the cache. We are overwriting the cache.

> Why is the network-only policy required?

Changing the `key` prop when auth changes re-renders the content without clearing the cache. For certain views we want to make sure the graphql query doesn't use a cached response when re-rendering, we add the `network-only` policy.

Reply:
> We're not clearing the cache. We are overwriting the cache.

I don't understand that distinction..

Reply:
Consider the scenario where a user is on https://taskcluster-web.netlify.com/secrets/garbage%2Farmenzg%2FfakeOrg.

Using `clearStore`:
* User is able to see the secret.
* User logs out.
* `clearStore` will cause the store to be cleared.
* The user still sees the secret.

Using `resetStore`:
* User is able to see the secret.
* User logs out.
* `resetStore` will cause the store to be cleared and all active queries to be refetched.
* User sees an error saying they are missing permissions to view the secret. This will work for the secrets view but will fail for the TaskGroup since all of the `fetchMore` queries won't be triggered therefore showing only the first 20 tasks https://github.com/taskcluster/taskcluster/blob/f283feaacb60b06d47c2a24344e857f9a2c7d217/ui/src/views/Tasks/TaskGroup/index.jsx#L52

Using the `key` prop and the `network-only` policy:
* User is able to see the secret.
* User logs out.
* The `key` prop will change. This will reconstruct the view from scratch (i.e., the constructor will run again; this is different than just re-running the render method). Changing the key will have the same effect as reloading the page but in this case the page is the view component.

* The graphql query of that view will see that it has `network-only` as a policy so it will always make a request to the server. The response will overwrite whatever was in the store for that query.
* User sees an error saying they are missing permissions to view the secret.

Reply:
OK, got it -- so `clearStore` and `key` + `network-only` are two different layers of protection here.